File: src/main/java/com/nicomincuzzi/frameworkless/maze/Room.java
package com.nicomincuzzi.frameworkless.maze;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

import java.util.List;
import java.util.Optional;

@Data
public class Room {
    @JsonProperty
    private int id;

    @JsonProperty
    private String name;

    @JsonProperty
    private Integer north;

    @JsonProperty
    private Integer south;

    @JsonProperty
    private Integer east;

    @JsonProperty
    private Integer west;

    @JsonProperty
    private List<Utensil> objects;

    public Optional<Room> getRoomById(int roomId, List<Room> rooms) {
        return rooms.stream().filter(x -> x.getId() == roomId).findFirst();
    }
}

File: src/main/kotlin/com/nicomincuzzi/frameworkless/maze/Room.kt
package com.nicomincuzzi.frameworkless.maze

import com.fasterxml.jackson.annotation.JsonProperty
import java.util.*

data class Room(
        @JsonProperty
        val id: Int = -1,

        @JsonProperty
        val name: String = "",

        @JsonProperty
        var north: Int? = null,

        @JsonProperty
        var south: Int? = null,

        @JsonProperty
        var east: Int? = null,

        @JsonProperty
        var west: Int? = null,

        @JsonProperty
        val objects: List<Utensil> = listOf()
) {
    fun getRoomById(roomId: Int, rooms: List<Room>): Optional<Room> {
        return rooms.stream().filter { x: Room -> x.id == roomId }.findFirst()
    }
}